src/status.rs

```diff
@@ -214,9 +214,23 @@ impl Status {
         header_map.insert(GRPC_STATUS_HEADER_CODE, self.code.to_header_value());
 
         if !self.message.is_empty() {
-            header_map.insert(GRPC_STATUS_MESSAGE_HEADER, HeaderValue::from_str(&self.message)
-                .map_err(invalid_header_value_byte)?);
+            let good_bytes = &self.message.as_bytes().iter().all(|&x| DEFAULT_ENCODE_SET.contains(x));
+            let to_write = 
+            match good_bytes{
```

Comment:
The name `good_bytes` might not give the best description. To me, this reads as "is the message all good_bytes? yes: percent encode them. no: send as-is". We'd want the opposite sentence.

Now, the tests appear to pass, so the *code* is right. We just want to fix up the narrative. I'd change the variable name to `needs_encode` or similar.

And then, since this is a boolean, it's more idiomatic in Rust to just use `if`, like `if needs_encoded { ... } else { ... }`.